tap_pardot/streams.py

```diff
@@ -106,6 +108,15 @@ def sync(self):
                 for rec in self.sync_page():
                     records_synced += 1
                     yield rec
+
+        except InvalidCredentials as e:
+            LOGGER.error(
+                "exception: %s \n traceback: %s",
+                e,
+                traceback.format_exc(),
+            )
+            return sys.exit(5)
```

Comment:
Note that sys.exit simply raises an exception, and thus doesn't ever return. I'd suggest removing the return here, since it's never reached, just to not make any confusion about the signature of this function.

https://docs.python.org/3/library/sys.html#sys.exit

Reply:
I actually added the return because I saw the return sys.exit(1) on the Exception.